script.js

```diff
@@ -24,6 +14,121 @@ var data = [
+  { label: '1x Coca Cola', value: 1, xp: '10' },
+  { label: '1x Tonic Water', value: 1, xp: '10' },
+  { label: '1x Ginger Beer', value: 1, xp: '11' },
+  { label: '1x Pineapple Juice', value: 1, xp: '12' },
+  { label: '1x Orange Juice', value: 1, xp: '13' },
+];
+
+var shots = [
+  { label: '2x Tequila', value: 1, xp: '2' },
+  { label: '1x Jägermeister', value: 1, xp: '4', },
+  { label: '1x Sambuca', value: 1, xp: '1', },
+  { label: '1x Pure Apple', value: 1, xp: '5' },
+  { label: '1x Pure Liquorice', value: 1, xp: '5' },
+  { label: '1x Råstof Caramel', value: 1, xp: '5' },
+  { label: '1x Råstof Rhubarb', value: 1, xp: '5' },
+  { label: '1x Rum', value: 1, xp: '3' },
+  { label: '1x Gin', value: 1, xp: '9' },
+  { label: '2x Fernet', value: 1, xp: '10' },
+  { label: '2x Mintuu', value: 1, xp: '12' },
+  { label: '1x Fireball', value: 1, xp: '13' },
+]
+
+var sirup = [
+  { label: '1x Mango', value: 1, xp: '2' },
+  { label: '1x Passionfruit', value: 1, xp: '2' },
+  { label: '1x Lime', value: 1, xp: '2' },
+  { label: '1x Sugar', value: 1, xp: '2' },
+  { label: '1x Grenadine', value: 1, xp: '2' },
+  { label: '2x Mango', value: 1, xp: '2' },
+  { label: '2x Passionfruit', value: 1, xp: '2' },
+  { label: '2x Lime', value: 1, xp: '2' },
+  { label: '2x Sugar', value: 1, xp: '2' },
+  { label: '2x Grenadine', value: 1, xp: '2' },
+  { label: '3x Mango', value: 1, xp: '2' },
+  { label: '3x Passionfruit', value: 1, xp: '2' },
+]
+
+// Init data filled in the wheel
+var data = og;
+
+// Different options for the wheel
+var options = {
+  "og": og,
+  "mixers": mixers,
+  "shots": shots,
+  "sirup": sirup
+};
+
+var menuItem = document.getElementById('wheelOptions');
+
+// Fill menu with options
+for (var option in options) {
+  var opt = document.createElement('option');
+  opt.value = option;
+  opt.innerHTML = option;
+  menuItem.appendChild(opt);
+}
+
+// Update wheel based option chosen
+menuItem.addEventListener('change', updateWheel);
+function updateWheel(event) {
+  var option = event.target.value;
+  data = options[option];
+
+  vis.selectAll('.slice').remove();
+  var arcs = vis.selectAll('g.slice').data(pie).enter().append('g').attr('class', 'slice');
+
+  arcs.append('path').attr('fill', function (d, i) {
+    return color(i);
+  })
+  .attr('d', function (d) {
+    return arc(d);
+  });
+
+  arcs
+    .append('text')
+    .attr('transform', function (d) {
+      d.innerRadius = 0;
+      d.outerRadius = r;
+      d.angle = (d.startAngle + d.endAngle) / 2;
+      return 'rotate(' + (((d.angle * 180) / Math.PI - 90) + 3) + ')translate(' + (d.outerRadius - 20) + ')';
+    })
+    .attr('font-size', '16')
+    .attr('stroke', 'black')
+    .attr('font-align', 'center')
+    .attr('stroke-width', 0.6)
+    .attr('fill', function(d,i){
+      return (i % 2 == 1) ? "black" : "white";
+    })
+    .attr('text-anchor', 'end')
+    .text(function (d, i) {
+      return data[i].label;
+    });

```

Comment:
This looks to me to be duplicate of existing code further down in the file used to render the wheel itself with new options. Can we make it into a separate function and call the same function both places to ensure consistency?

Reply:
Resolved